hack/feature-deploy.sh

```diff
@@ -2,11 +2,67 @@
 
 set -e
 
+if [ "$FEATURES_ENVIRONMENT" == "" ]; then
+	echo "[ERROR]: No FEATURES_ENVIRONMENT provided"
+	exit 1
+fi
+
+if [ "$FEATURES" == "" ]; then
+	echo "[ERROR]: No FEATURES provided"
+	exit 1
+fi
+
 # expect oc to be in PATH by default
-OC_TOOL="${OC_TOOL:-oc}"
+export OC_TOOL="${OC_TOOL:-oc}"
+
+# expect kustomize to be in PATH by default
+KUSTOMIZE="${KUSTOMIZE:-kustomize}"
+
+# Deploy features
+success=0
+iterations=0
+sleep_time=10
+max_iterations=30 # results in 5 minute timeout
+until [[ $success -eq 1 ]] || [[ $iterations -eq $max_iterations ]]
+do
+
+  feature_failed=0
+
+  for feature in $FEATURES; do
+
+    feature_dir=feature-configs/${FEATURES_ENVIRONMENT}/${feature}/
+    if [[ ! -d $feature_dir ]]; then
+      echo "[WARN] Feature '$feature' is not configured for environment '$FEATURES_ENVIRONMENT', skipping it"
+      continue
+    fi
+
+    echo "[INFO] Deploying feature '$feature' for environment '$FEATURES_ENVIRONMENT'"
+    set +e
+    if ! ${KUSTOMIZE} build $feature_dir | ${OC_TOOL} apply -f -
+    then
+      echo "[WARN] Deployment of feature '$feature' failed."
+      feature_failed=1
+    fi
+    set -e
+
+  done
```

Comment:
I think this logic is incorrect:
1. afaik apply will return 0 regardless of other errors that might occur
2. even if such errors occur, there's no need to re-apply, just wait for them to settle. 
3. which this loop doesnt do (something like kubectl rollout status, which isn't available in oc[??] not sure whats the alternative). ie we would probably be out of that loop before features declarations are really "done"

other then that, we can use a single oc apply -k, if we create a folder referencing all the wanted features.
which IMHO would be 'cleaner' than running multiple kustomize commands. we can even have an overlay dir with some supported/most relevant variations (all, networking, performance, etc)

Reply:
Thx for review!

1. no, this works: as long as the CR can't be posted because the CRD isn't there yet, it returns an error
2. we don't want custom wait logic, that's why we iterate until all succeeds. Custom wait would be "check if CRD exists already"
3. yes, we might want to have a sanity check that everything works as expected. I'd like to leave that out of scope of this PR, in order to get it merged asap, so that others can add more  features on top of this
4. about `oc -k`, see https://github.com/openshift-kni/cnf-features-deploy/pull/4#discussion_r366218426